authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationHDFSPlugin.java

```diff
@@ -0,0 +1,257 @@
+                      rangerSecurableObjects.add(
+                          generateAuthorizationSecurableObject(
+                              translateMetadataObject(securableObject).names(),
+                              RangerHDFSMetadataObject.Type.PATH,
+                              rangerPrivileges));
+                      break;
+                    default:
+                      throw new AuthorizationPluginException(
+                          "The privilege %s is not supported for the securable object: %s",
+                          gravitinoPrivilege.name(), securableObject.type());
+                  }
+                  break;
+                default:
+                  LOG.warn(
+                      "RangerAuthorizationHDFSPlugin -> privilege {} is not supported for the securable object: {}",
+                      gravitinoPrivilege.name(),
+                      securableObject.type());
+              }
+            });
+
+    return rangerSecurableObjects;
+  }
+
+  @Override
+  public List<AuthorizationSecurableObject> translateOwner(MetadataObject gravitinoMetadataObject) {
+    List<AuthorizationSecurableObject> rangerSecurableObjects = new ArrayList<>();
+    switch (gravitinoMetadataObject.type()) {
+      case METALAKE:
+      case CATALOG:
+      case SCHEMA:
+        return rangerSecurableObjects;
+      case FILESET:
+        rangerSecurableObjects.add(
+            generateAuthorizationSecurableObject(
+                translateMetadataObject(gravitinoMetadataObject).names(),
+                RangerHDFSMetadataObject.Type.PATH,
+                ownerMappingRule()));
+        break;
+      default:
+        throw new AuthorizationPluginException(
+            "The owner privilege is not supported for the securable object: %s",
+            gravitinoMetadataObject.type());
+    }
+
+    return rangerSecurableObjects;
+  }
+
+  @Override
+  public AuthorizationMetadataObject translateMetadataObject(MetadataObject metadataObject) {
+    Preconditions.checkArgument(
+        allowMetadataObjectTypesRule().contains(metadataObject.type()),
+        String.format(
+            "The metadata object type %s is not supported in the RangerAuthorizationHDFSPlugin",
+            metadataObject.type()));
+    List<String> nsMetadataObject =
+        Lists.newArrayList(SecurableObjects.DOT_SPLITTER.splitToList(metadataObject.fullName()));
+    Preconditions.checkArgument(
+        nsMetadataObject.size() > 0, "The metadata object must have at least one name.");
+
+    if (metadataObject.type() == MetadataObject.Type.FILESET) {
+      RangerHDFSMetadataObject rangerHDFSMetadataObject =
+          new RangerHDFSMetadataObject(
+              getFileSetPath(metadataObject), RangerHDFSMetadataObject.Type.PATH);
+      rangerHDFSMetadataObject.validateAuthorizationMetadataObject();
+      return rangerHDFSMetadataObject;
+    } else {
+      return new RangerHDFSMetadataObject("", RangerHDFSMetadataObject.Type.PATH);
+    }
+  }
```

Comment:
@theoryxu @xunliu 
As suggested, could you create an interface to handle this logic? for example `interface PathBasedAuthorization extends AuthorizationPlugin` to make it reusable, as cloud IAM authorization also needs it to handle fileset authorization.

Reply:
Please consider if it's reasonable.

Reply:
I'm sorry, I don't seem to get it.

`translateMetadataObject ` is a method from `AuthorizationPrivilegesMappingProvider`

IAM authorization will also implement the `AuthorizationPrivilegesMappingProvider` , right?

Reply:
@yuqi1129 I think we can add interface `PathBasedAuthorization` in your next `IAMAuthorizationPlugin` PR. This PR didn't need modify.